_posts/current/blog/2022-03-01-how-to-build-flang-on-windows-on-arm.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: post
+title: How to build flang on Windows on Arm
+description: "In this blog post we look at how to build flang on Windows on Arm. "
+date: 2022-03-01 09:38:26 +00:00
+image: /assets/images/content/careers-image-1.jpg
+tags:
+  - Windows on Arm
+  - Windows
+  - Flang
+  - LLVM
+  - Clang
+category: blog
+author: diana.picus
+---
+Last week we published a blog on [how to set up Windows on Arm for LLVM development](https://www.linaro.org/blog/how-to-set-up-windows-on-arm-for-llvm-development/). In this blog we provide a step-by-step guide on how to build Flang on Windows on Arm.
+
+## What is Flang?
+
+Flang is a Fortran front-end developed as part of the LLVM project. It is still a young project and Linaro is contributing to its growth on both Linux and Windows on Arm.
+
+If you need to build Fortran code on Windows on Arm, there really isn't any battle-tested native open source compiler to rely on. We are hoping to help flang fill that void.
+
+Naturally, the first step is to be able to compile flang itself on Windows on Arm, and this is exactly what this blog post will cover. 
+
+## How to build Flang
+
+In order to get a build of flang, you will first need to setup your machine for LLVM development as described in [our previous blog post ](https://www.linaro.org/blog/how-to-set-up-windows-on-arm-for-llvm-development/)mentioned above.
+
+You will have to add flang, clang and mlir to the list of enabled projects.
+
+You will also need to tell clang where to find the builtins, otherwise the linker will complain that it cannot find symbols such as _udivdi3 or others. Adding the path to the builtin library as described in the [Clang Compiler User’s Manual — Clang 13 documentation](https://clang.llvm.org/docs/UsersManual.html#finding-clang-runtime-libraries) doesn’t seem to be working at the moment (bug report pending). Luckily, we can work around this issue by adding the builtin library directly on every link command 
+
+```
+*(-DCMAKE**_LINKER_FLAGS=path/to/clang_rt.builtins-aarch64.lib;
```

Comment:
Is there a way to make this look like inline code? If not, I think it would be better to just leave as plain text.
Also, there are some spurious * and a missing underscore; it's supposed to be:
```suggestion
(-DCMAKE_*_LINKER_FLAGS=path/to/clang_rt.builtins-aarch64.lib;
```